src/Microsoft.ML.Data/DataView/CompositeRowToRowMapper.cs

```diff
@@ -70,7 +75,11 @@ public Row GetRow(Row input, Func<int, bool> active)
             var deps = new Func<int, bool>[InnerMappers.Length];
             deps[deps.Length - 1] = active;
             for (int i = deps.Length - 1; i >= 1; --i)
-                deps[i - 1] = InnerMappers[i].GetDependencies(deps[i]);
+            {
+                var outputColumns = InnerMappers[i].OutputSchema.Where(c => deps[i](c.Index));
+                var cols = InnerMappers[i].GetDependencies(outputColumns);
```

Comment:
>GetDependencies [](start = 43, length = 15)

I would put ToArray to cache it. Otherwise you constantly fetch results from IEnumerable